### PR TITLE
build(typescript-eslint): configure typescript-eslint package

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

@splincode As you remember we discussed that `tslint` is goint to be deprecated in 2019. We've got to prepare the ground for the migration from `tslint --> typescript-eslint`. Currently we shouldn't remove the `tslint` package at all, we can keep that combo of `tslint + typescript-eslint` and remove `tslint` after its completely deprecated.

### CODE REVIEW

**Comment by arturovt** [comment]
@splincode cc

**Comment by splincode** [comment]
```
$ yarn lint && yarn eslint
```

Do they do the same thing?

**Comment by arturovt** [comment]
> ```
> $ yarn lint && yarn eslint
> ```
> 
> Do they do the same thing?

Nope, `lint` for `tslint` and `eslint` for `eslint`.

**Comment by arturovt** [comment]
There is a need to decide if we're ready to reject `tslint` right now, but we will have to fix `eslint` issues.

**Comment by arturovt** [comment]
I turned them off in the `rules` section

**Comment by splincode** [comment]
@markwhitfeld I am ready to give up tslint

**Comment by arturovt** [comment]
@splincode I guess we could do it iteratevily via small steps, what do you think?

**Comment by arturovt** [comment]
@markwhitfeld ping

**Comment by markwhitfeld** [comment]
I agree with the iterative approach. It also gives us the chance to review the rules one by one and discuss if necessary.
Let me review now...

File: .circleci/config.yml [hunk 1]
```diff
@@ -85,7 +85,9 @@ jobs:
       - *attach_workspace
       - run:
           name: Lint
-          command: yarn lint
+          command: |
+            yarn lint
+            yarn eslint
```
Comment (markwhitfeld) [inline .circleci/config.yml]: Could we do these as parallel circle CI tasks?
(just like how the tests are as separate circle ci tasks and therefore get run in parallel)

Comment (arturovt) [reply .circleci/config.yml]: Yep, all changes were made by eslint. TSLint also has [this rule](https://palantir.github.io/tslint/rules/interface-over-type-literal/)

Comment (arturovt) [reply .circleci/config.yml]: > Could we do these as parallel circle CI tasks?
> (just like how the tests are as separate circle ci tasks and therefore get run in parallel)

Sure

Comment (markwhitfeld) [reply .circleci/config.yml]: Thanks!

**Comment by arturovt** [comment]
> Broken build to fix?

:joy: Gonna fix it

**Comment by arturovt** [comment]
@markwhitfeld ping

**Pull request merged**